src/System.Runtime.Analyzers/Core/CallGCSuppressFinalizeCorrectly.cs

```diff
@@ -27,42 +29,239 @@ public abstract class CallGCSuppressFinalizeCorrectlyAnalyzer : DiagnosticAnalyz
                                                                              s_localizableMessageNotCalledWithFinalizer,
                                                                              DiagnosticCategory.Performance,
```

Comment:
will do. I merged after speaking with manish offline, will have to send another PR :\